File: subprojects/frida-python/examples/enumerate_processes.py
from pprint import pformat

from pygments import highlight
from pygments.formatters import Terminal256Formatter
from pygments.lexers import PythonLexer

import frida

device = frida.get_usb_device()


def trim_icon(icon):
    result = dict(icon)
    result["image"] = result["image"][0:16] + b"..."
    return result


processes = device.enumerate_processes(scope="full")
for proc in processes:
    params = dict(proc.parameters)
    if "icons" in params:
        params["icons"] = [trim_icon(icon) for icon in params["icons"]]
    print(
        f'Process(pid={proc.pid}, name="{proc.name}", parameters={highlight(pformat(params), PythonLexer(), Terminal256Formatter()).rstrip()})'
    )

File: subprojects/frida-node/examples/enumerate_processes.ts
import frida from "frida";
import { inspect } from "util";

async function main() {
    const device = await frida.getUsbDevice();
    const processes = await device.enumerateProcesses({ scope: frida.Scope.Full });
    console.log("[*] Processes:", inspect(processes, {
        maxArrayLength: 500,
        depth: 4,
        colors: true
    }));
}

main()
    .catch(e => {
        console.error(e);
    });
